app/services/container_topology_service.rb

```diff
@@ -47,8 +47,36 @@ def build_topology
     topology
   end
 
-  def build_entity(id, name, kind)
-    {:metadata => {:id => id, :name => name}, :kind => kind}
+  def build_entity_data(entity, kind)
+    status = entity_status(entity, kind)
+
+    case kind
+    when 'VM', 'Host'
+      id = entity.uid_ems
+    else
+      id = entity.ems_ref
+    end
```

Comment:
> Ok, I have not checked Rubocop, I just guessed. There probably is not a rule for this in rubocop.
> 
> In ruby almost everything is an expression. The positive and the negative branches of a condition in this case the branches of the case/when statement.
> 
> In both the branches here, you are just assigning to id so you shall pull out (don't know if this is the correct mathematical term in English) the assignment in front of the case statement. That improves readability, because it tells the reader of your code that the whole purpose of your case statement is the calculation of the id value and there should be no other effects. It also simplifies further possible refactoring, such as extracting a function.
> 
> See e.g. here: http://www.techotopia.com/index.php/The_Ruby_case_Statement except we align the case and when statements.

I reuploaded a revised version, please check it.
(BTW most of the code in ManageIQ doesn't comply with this writing style, so with lack of Rubocop and consistency in the codebase, it's really hard to know that's what it should look like)